pkg/reconcile/pipeline/infinispan/handler/manage/conditions.go

```diff
@@ -41,22 +41,21 @@ func PreliminaryChecks(i *ispnv1.Infinispan, ctx pipeline.Context) {
 	}
 }
 
+var cacheServiceForbiddenMsg = "Please change to service type DataGrid. CacheService is no more supported."
+
 func OperatorStatusChecks(i *ispnv1.Infinispan, ctx pipeline.Context) {
 	operatorPod := kube.GetOperatorPodName()
 	// Pod name is changed, means operator restarted
 	if i.Status.Operator.Pod != operatorPod {
-		if i.IsCache() {
-			errMsg := "ServiceType Cache configured is deprecated and will be removed in future releases. ServiceType DataGrid is recomended."
-			ctx.EventRecorder().Event(i, corev1.EventTypeWarning, "DeprecatedCacheService", errMsg)
-			ctx.Log().Info(errMsg)
+		if i.IsCache() && ctx.Operand().BansCacheService() {
```

Comment:
I think this should be outside of the pod if check as we always want to set `WellFormed=False` if the cache service is present. Furthermore, we need to ensure that if cache service is present that we don't continue with the reconcilliation pipeline by calling `ctx.Stop(nil)`